central/activecomponent/datastore/internal/store/postgres/store.go

```diff
@@ -93,11 +93,13 @@ func metricsSetAcquireDBConnDuration(start time.Time, op ops.Op) {
 	metrics.SetAcquireDBConnDuration(start, op, storeName)
 }
 
-func insertIntoActiveComponents(batch *pgx.Batch, obj *storage.ActiveComponent) error {
+func insertIntoActiveComponents(batch *pgx.Batch, pool pgSearch.BufferPool, obj *storage.ActiveComponent) (*[]byte, error) {
 
-	serialized, marshalErr := obj.MarshalVT()
+	buf := pool.Get(obj.SizeVT())
```

Comment:
Interesting would be to see if the new [Green Tea GC](https://go.dev/blog/greenteagc) achieves the same performance.

Reply:
Yeah, I was wondering how we can measure the difference. For example change to swiss map went unnoticed at least by me